earth_enterprise/src/fusion/gst/gstAssetGroup.cpp

```diff
@@ -18,26 +18,43 @@
 #include <Qt/qregexp.h>
 #include <gstAssetManager.h>
 #include <khConstants.h>
-
 #include "gstAssetGroup.h"
 
+
 const char* shortAssetName(const char* n) {
-  QString sname(n);
-  QStringList suffix;
-  suffix << kVectorAssetSuffix.c_str() << kImageryAssetSuffix.c_str()
-         << kMercatorImageryAssetSuffix.c_str() << kTerrainAssetSuffix.c_str()
-         << kVectorProjectSuffix.c_str() << kImageryProjectSuffix.c_str()
-         << kMercatorImageryProjectSuffix.c_str() << kTerrainProjectSuffix.c_str()
-         << kDatabaseSuffix.c_str() << kMapLayerSuffix.c_str() << kMapProjectSuffix.c_str()
-         << kMapDatabaseSuffix.c_str() << kMercatorMapDatabaseSuffix.c_str()
-         << kVectorLayerSuffix.c_str();
-  for (QStringList::Iterator it = suffix.begin(); it != suffix.end(); ++it) {
-    if (sname.endsWith(*it)) {
-      sname.truncate(sname.length() - it->length());
-      break;
-    }
+
+  static const std::vector<std::string> suffixes =
+  {
+      kVectorAssetSuffix, kImageryAssetSuffix, kMercatorImageryAssetSuffix,
+      kTerrainAssetSuffix, kVectorProjectSuffix, kImageryProjectSuffix,
+      kMercatorImageryProjectSuffix, kTerrainProjectSuffix,
+      kDatabaseSuffix, kMapLayerSuffix, kMapProjectSuffix,
+      kMapDatabaseSuffix, kMercatorMapDatabaseSuffix, kVectorLayerSuffix
+  };
+
+  std::string saname { n };
+
+  for (const auto& elem : suffixes)
```

Comment:
I'm having trouble understanding what this code is doing, so can you add some comments?

Reply:
done